feat: multiple versions for the `timescaledb-apache` extension

yvan-sraka opened the pull request:
Build multiple versions of the `timescaledb-apache` extension on PostgreSQL 15.

yvan-sraka committed:
feat: multiple versions for the timescaledb-apache extension

yvan-sraka committed:
chores: opted-out nix-fast-build

yvan-sraka committed:
chores: set up nix-github-actions

yvan-sraka committed:
chores: comment out vestigal testinfra-ami-build

yvan-sraka committed:
chores: use custom github runners

samrose committed:
feat: multiversion timescaledb

samrose committed:
fix: refactor a bit to complete the extension

samrose committed:
test: remove older version of integration test

samrose committed:
fix: make sure integration tests
do not run timescaledb tests on pg > 15

samrose committed:
test: revert to use test in nix/tests for timescale

jfroche commented on nix/ext/tests/default.nix:
```diff
@@ -202,7 +186,7 @@ builtins.listToAttrs (
       "pg_cron"
       "pg_net"
-      "timescaledb-apache"
+      "timescaledb"
```
We can remove that from this list. We don't want to test timescaledb with the default nixos test

jfroche commented on nix/tests/timescaledb.nix:
```diff
@@ -0,0 +1,100 @@
+                  installed_version = run_sql(r"""SELECT extversion FROM pg_extension WHERE extname = 'timescaledb';""")
+                  assert installed_version == version, f"Expected timescaledb version {version}, but found {installed_version}"
+
+          start_all()
+
+          server.wait_for_unit("multi-user.target")
+          server.wait_for_unit("postgresql.service")
+
+          check_upgrade_path()
```
This should be indented at the same level as the def otherwise the test won't do anything.

samrose committed:
fix: integrtion tests

jfroche committed:
Move timescaledb nixos test to nix/ext/tests

samrose committed:
chore: bump for test

samrose committed:
fix: need to also handle timescaledb-loader

samrose committed:
chore: bump version for fix

jfroche committed:
feat: switch both timescaledb and its loader when switching versions

Also reuse default tests defined in lib.py to test extension upgrade
paths and switching extension versions.

samrose committed:
fix: didn't need "loader"

samrose committed:
chore: bump version for testing

samrose committed:
fix: does need loader

samrose committed:
chore: bump to test

samrose commented on nix/ext/timescaledb.nix:
```diff
@@ -98,6 +100,16 @@ let
+    runtimeInputs = [ coreutils ];
+    text = ''
+      EXT_LOADER_TO_USE="$EXT_WRAPPER_LIB/$EXT_NAME-loader-$VERSION${postgresql.dlSuffix}"
+      if [ -f "$EXT_LOADER_TO_USE" ]; then
```
unfortunately we cannot link to `EXT_LOADER_TO_USE="$EXT_WRAPPER_LIB/$EXT_NAME-loader-$VERSION${postgresql.dlSuffix}"` 

Because timescale will look for the loader at `timescaledb.so` when it starts so we have to link the versioned loader to timescaledb.so

samrose replied:
trying to resolve with https://github.com/supabase/postgres/pull/1749/commits/c7604f1aa810aa8626103048f1ee1368a4ef3bae

samrose committed:
fix: must symlink to timescaledb.so on switch

samrose committed:
chore: bump for release
